app/src/main/java/io/apicurio/registry/ArtifactIdGenerator.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright 2019 Red Hat
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.apicurio.registry;
+
+import java.util.UUID;
+
+import javax.enterprise.context.ApplicationScoped;
+
+/**
+ * @author eric.wittmann@gmail.com
+ */
+@ApplicationScoped
+public class ArtifactIdGenerator {
+
+    public String generate() {
```

Comment:
Yeah I'm not married to this - it might be better to let the storage layer generate the artifact ID.  However, this does have the nice benefit that we could plug in various different strategies for generating IDs and it would be independent of the storage provider.  I'm not sure those are good reasons or not.  We can easily change it going forward.  I'll keep this for now and we can rip it out if we all agree it's dumb. :)